stdpopsim/catalog/SusScr/demographic_models.py

```diff
@@ -0,0 +1,3462 @@
+import msprime
+import stdpopsim
+from . import species
+
+_species = stdpopsim.get_species("SusScr")
+
+
+def _WildBoar_4Z22():
+    id = "WildBoar_4Z22"
+    description = "Piecewise size model for wild boar (Zhang et al. 2022)"
+    long_description = """
+    This demographic model is a piecewise size model for wild boar
+    from Zhang et al. (2022). Effective population sizes were estimated by
+    PSMC with parameters Tmax = 20, n = 64 (4 + 50*1 + 4 + 6)
+    """
+    populations = [
+        stdpopsim.Population(id="EUW", description="European wild boar"),
+        stdpopsim.Population(id="NCW", description="North Chinese wild boar"),
+        stdpopsim.Population(id="SCW", description="South Chinese wild boar"),
+        stdpopsim.Population(id="SMW", description="Sumatran wild boar"),
+    ]
+    citations = [species._ZhangEtAl]
+    return stdpopsim.DemographicModel(
+        id=id,
+        description=description,
+        long_description=long_description,
+        populations=populations,
+        citations=citations,
+        generation_time=3,
+        mutation_rate=3.6e-9,
+        population_configurations=[
+            stdpopsim.PopulationConfiguration(
```

Comment:
@AprilYUZhang there is a lot of code here due to many timesteps. I suggest you follow the style from https://github.com/popsim-consortium/stdpopsim/blob/main/stdpopsim/qc/BosTau.py